### ISSUE TITLE
Compatibility with CSVW

### ISSUE THREAD

**Issue by randomstuff** [turicas/rows#118]
The [CSV on the Web WG](http://www.w3.org/2013/csvw/wiki/Main_Page) does a few interesting things such as defining a [parsing options for CSV-like formats](http://www.w3.org/TR/2015/WD-tabular-data-model-20150416/#parsing) and defining [JSON-based dialect description](http://www.w3.org/TR/2015/WD-tabular-metadata-20150416/#dfn-dialect-description). It would be nice to handle this.

**Comment by randomstuff** [comment]
This is related to #36 as JSON Table Schema is the ancestor of CSVW AFAIU.

**Comment by turicas** [comment]
@randomstuff, thanks! Do you know any Python library that implements these standards?

**Comment by randomstuff** [comment]
I looked in the [tools](http://www.w3.org/2013/csvw/wiki/Tools) section but didn't find anything relevant. I guess it still needs to be written.

By the way, it might be interesting to manage to get rows on this list.

**Comment by randomstuff** [comment]
Related to #80 as well.

**Comment by turicas** [comment]
@randomstuff, as #80 was implemented, `rows` now automatically detects the CSV dialect, and it can be specified as a parameter also (if you don't want it to be detected). We still need to implement encoding detection (#129).

When encoding detection is implemented we may create a function `rows.plugins.csv.create_description` to generate a `dict` compatible with this JSON-based dialect description (and use it internally on `rows.plugins.csv.import_from_csv` also). Then, we could just add a parameter to do not use the internal `create_description` function but read it from a JSON file or Python `dict` instead (the parameter could be the same, actually: `dialect`).

BTW, do you know how we can get `rows` in the tools section?

**Comment by randomstuff** [comment]
No, I don't know how you can get `rows` into the tools ection. Maybe joigning the WG might help?
